sklearn/utils/tests/test_array_api.py

```diff
@@ -204,6 +355,8 @@ def test_nan_reductions(library, X, reduction, expected):
     with config_context(array_api_dispatch=True):
         result = reduction(xp.asarray(X))
 
+    result = _convert_to_numpy(result, xp)
```

Comment:
I think some of the NaN reduction test failures is because of this added line. If we remove L352 and L356 as well then the tests related to Cupy errors pass again. I'm not sure which option is better: (a) remove the `_convert_to_numpy` or (b) remove the two lines above

Reply:
I think we should just skip tests that are related to an upstream cupy bug. Do you have the URL handy?

Reply:
Actually, the problem is that expected should be a numpy array as well if we want to compare the two using `assert_allclose`.